coderd/database/dbgen/dbgen.go

```diff
@@ -240,6 +240,14 @@ func User(t testing.TB, db database.Store, orig database.User) database.User {
 		})
 		require.NoError(t, err, "user last seen")
 	}
+
+	if orig.Deleted {
+		err = db.UpdateUserDeletedByID(genCtx, database.UpdateUserDeletedByIDParams{
```

Comment:
FYI There is also dormant status, hopefully it does not change anything.

Reply:
All combinations of login_type, status, and deleted should be covered by this test.

Reply:
I'm fairly sure user dormancy and deletion are completely orthogonal i.e. both can be set independently.